sdk/digitaltwins/digitaltwins/src/digitalTwinsClient.ts

```diff
@@ -0,0 +1,898 @@
+    options: RequestOptionsBase
+  ): AsyncIterableIterator<any> {
+    for await (const page of this.listRelationshipsPage(digitalTwinId, options, {})) {
+      const value = page.value || [];
+      for (const item of value) {
+        yield item;
+      }
+    }
+  }
+
+  /**
+   * Retrieve relationships for a digital twin.
+   *
+   * @param digitalTwinId The Id of the digital twin.
+   */
+  public listRelationships(
+    digitalTwinId: string,
+    options: RequestOptionsBase & PageSettings = {}
+  ): PagedAsyncIterableIterator<any, DigitalTwinsListRelationshipsResponse> {
+    const iter = this.listRelationshipsAll(digitalTwinId, options);
+
+    return {
+      next() {
+        return iter.next();
+      },
+      [Symbol.asyncIterator]() {
+        return this;
+      },
+      byPage: (settings: PageSettings = {}) =>
+        this.listRelationshipsPage(digitalTwinId, options, settings)
+    };
+  }
+
+  /**
+   * @internal
+   * @ignore
+   * Deals with the pagination of {@link listIncomingRelationships}.
+   *
+   * @param {PageSettings} continuationState An object that indicates the position of the paginated request.
+   * @param {RequestOptionsBase} [options] Common options for the iterative endpoints.
+   *
+   */
+  private async *listIncomingRelationshipsPage(
+    digitalTwinId: string,
+    options: RequestOptionsBase,
+    continuationState: PageSettings
+  ): AsyncIterableIterator<DigitalTwinsListIncomingRelationshipsResponse> {
+    if (continuationState.continuationToken == null) {
+      const optionsComplete: RequestOptionsBase = {
+        maxresults: continuationState.maxPageSize,
+        ...options
+      };
+      const listIncomingRelationshipsResponse = await this.client.digitalTwins.listIncomingRelationships(
+        digitalTwinId,
+        optionsComplete
+      );
+      continuationState.continuationToken = listIncomingRelationshipsResponse.nextLink;
+      yield listIncomingRelationshipsResponse;
+    }
+    while (continuationState.continuationToken) {
+      const listIncomingRelationshipsResponse = await this.client.digitalTwins.listIncomingRelationshipsNext(
+        continuationState.continuationToken,
+        "",
+        options
+      );
+
+      continuationState.continuationToken = listIncomingRelationshipsResponse.nextLink;
+      yield listIncomingRelationshipsResponse;
+    }
+  }
+
+  /**
+   * @internal
+   * @ignore
+   * Deals with the iteration of all the available results of {@link listIncomingRelationships}.
+   * @param {RequestOptionsBase} [options] Common options for the iterative endpoints.
+   */
+  private async *listIncomingRelationshipsAll(
+    digitalTwinId: string,
+    options: RequestOptionsBase
+  ): AsyncIterableIterator<IncomingRelationship> {
+    const f = {};
+
+    for await (const page of this.listIncomingRelationshipsPage(digitalTwinId, options, f)) {
+      const value = page.value || [];
+      for (const item of value) {
+        yield item;
+      }
```

Comment:
I don't think you need to do this, you can do something like

```suggestion
        yield* value;
```

This will yield the entire iterable without you having to loop over it and yield each member.

Reply:
Good point - changed.

---
In reply to: [470204871](https://github.com/Azure/azure-sdk-for-js/pull/10560#discussion_r470204871) [](ancestors = 470204871)